### PR TITLE
xbps-fetch: TIMEOUT_CONNECTION to customize time waiting for response

### PR DESCRIPTION
**Pull request by Chocimier**
Allows to [configure CI job](https://github.com/Chocimier/void-packages-org/commit/a09733b7748ede191a4cbf44bc447952d96bdf63) to build from ftp source without taking too much time as [demostrated with chrootkit](https://travis-ci.org/Chocimier/void-packages-org/builds/647493514).

### CODE REVIEW

**Comment by xtraeme** [comment]
ok looks good at first glance

@duncaen ?

**Comment by xtraeme** [comment]
```
=> chkrootkit-0.53_2: fetching distfile 'chkrootkit-0.53.tar.gz'...

ftp://ftp.pangeia.com.br/pub/seg/pac/chkrootkit-0.53.tar.gz: Can't open data connection

=> chkrootkit-0.53_2: fetch attempt 2 of 15...

chkrootkit-0.53.tar.gz: [40KB 10%] 33MB/s ETA: 00m00s

chkrootkit-0.53.tar.gz: 40KB [avg rate: 322MB/s]
```

FTR

File: bin/xbps-fetch/xbps-fetch.1 [hunk 1]
```diff
@@ -82,6 +82,9 @@ Overrides the default CA certificates path, by default set to
 Sets the SSL/TLS client verification certificate file.
 .It Sy SSL_CLIENT_KEY_FILE
 Sets the SSL/TLS client verification key file.
+.It Sy TIMEOUT_CONNECTION
+Sets connection timeout in milliseconds.
+When zero, default value is used.
```
Comment (xtraeme) [inline bin/xbps-fetch/xbps-fetch.1]: what's default value? I don't know

**Comment by Duncaen** [comment]
`CONNECTION_TIMEOUT` sounds more natural, this wouldn't allow 0 or -1 (infinite), I'll have to think about this tomorrow.

**Comment by Chocimier** [comment]
Variable rename, -1 allowed, default timeout documented.

**Comment by xtraeme** [comment]
I'm ok with this one, @Duncaen ?

**Comment by xtraeme** [comment]
The default seems to be too high?

**Comment by Duncaen** [comment]
I think that this is what curl used by default.

I guess the PR is fine, but I still think the xbps-src patch is wrong, depending on `xbps-fetch` behavior while everything else does not specifically depend on the fetch_cmd.

**Comment by Chocimier** [comment]
> xbps-src patch is wrong, depending on xbps-fetch behavior

Didn't think of that, but it's not that bad. If fetch_cmd happens to be xbps-fetch, some settings are applied, in other cases nothing happens. Given that there is no standard way, other options I see are: enumerating other supported tools or ignoring problem at all.
Proposals how to improve are welcome, naturally.

**Comment by xtraeme** [comment]
I still think that 5 minutes is too much. How about changing it to 1 minute?

**Comment by xtraeme** [comment]
Also while it's ok to have the retry code in xbps-src, I would like to have this in `xbps-fetch(1)` instead. This is already in `TODO`.

**Comment by xtraeme** [comment]
unless this gets any progress it will have to wait until 0.60, because I plan to release 0.59 tomorrow.

**Comment by Chocimier** [comment]
Using what curl uses sound ok, but I can change if you insist. But not the day before release.

**Comment by xtraeme** [comment]
If this is what curl uses OK, fine with me.

**Pull request merged**